pkgs/racket-doc/scribblings/reference/for.scrbl

```diff
@@ -91,6 +91,14 @@ unreachable, and if the @racket[for] body can no longer reference an
 constructor supports additional sequences that behave like lists and
 streams in this way.
 
+If a @racket[seq-expr] is a literal list, vector, exact integer,
+string, byte string, immutable hash, or expands to such a literal,
+then it may be treated as if a sequence transformer such as
+@racket[in-list] was used, unless the @racket[seq-expr] has a true
```

Comment:
Can you add "to improve performance" or something along that line to clarify why doing so is desirable?